### PR TITLE
Fix infinite loop in ThreadTracks when zooming-in a lot

### PR DESCRIPTION
**Pull request by karupayun**
We sometimes have an issue when zooming-in a lot. Orbit hangs because falls in an infinite loop while iterating through pixels in ThreadTracks.

After investigating a bit, I found that the issue is a regression from https://github.com/google/orbit/pull/4296. We are using 32-bits integer to get the pixel number, but the requested pixel (end_time of the current timer) could be very far away and as it overflows, Orbit ends iterating in an infinite loop.

So, in this PR I'm fixing this issue and updating the unit-tests since they should have caught the issue.

Test: New unit test fails with the old-code. Now they are green. Load a capture, see that now it doesn't hang anymore.

### CODE REVIEW

**Review by danielfenner — APPROVED** [review]
Thanks for fixing this! lgtm.

File: src/ClientData/FastRenderingUtilsTest.cpp [hunk 1]
```diff
@@ -41,9 +41,11 @@ TEST_P(GetNextPixelBoundaryTimeNsTest, TimestampAreInRange) {
     return 1 + (dividend - 1) / divisor;
   };
 
+  constexpr uint32_t kNumberOfTestedTimestamps = 200;
```
Comment (dpallotti) [inline src/ClientData/FastRenderingUtilsTest.cpp:44]: Unrelated: in the name of the test: TimestampAreInRange -> TimestampsAreInRange

File: src/ClientData/FastRenderingUtilsTest.cpp [hunk 2]
```diff
@@ -8,8 +8,8 @@
 
 namespace orbit_client_data {
 
-constexpr uint64_t kStartNs = 100;
-constexpr uint64_t kEndNs = 200;
+constexpr uint64_t kStartNs = 100'000'000'000'000ULL;
+constexpr uint64_t kEndNs = 200'000'000'000'000ULL;
```
Comment (dpallotti) [inline src/ClientData/FastRenderingUtilsTest.cpp:12]: In a way this changes a couple of the tests below quite a bit, as it changes `kEndNs - kStartNs` from 100 to 10^11. Is that ok?

**Review by dpallotti — APPROVED** [review]

File: src/ClientData/FastRenderingUtilsTest.cpp (hunk 2 above)
Comment (karupayun) [reply src/ClientData/FastRenderingUtilsTest.cpp:12]: Yes, It's actually better.

**Pull request merged**